File: JANUARY/16-01-2026/2975.cpp
class Solution {
public:
    int maximizeSquareArea(int m, int n, vector<int>& hFences, vector<int>& vFences) {
        const long long MOD = 1000000007;
        
        hFences.push_back(1);
        hFences.push_back(m);
        vFences.push_back(1);
        vFences.push_back(n);
        
        sort(hFences.begin(), hFences.end());
        sort(vFences.begin(), vFences.end());
        
        unordered_set<long long> hs;
        for (int i = 0; i < hFences.size(); i++)
            for (int j = i + 1; j < hFences.size(); j++)
                hs.insert((long long)hFences[j] - hFences[i]);
        
        long long best = -1;
        for (int i = 0; i < vFences.size(); i++) {
            for (int j = i + 1; j < vFences.size(); j++) {
                long long d = (long long)vFences[j] - vFences[i];
                if (hs.count(d)) best = max(best, d);
            }
        }
        
        if (best == -1) return -1;
        return (best % MOD) * (best % MOD) % MOD;
    }
};

File: JANUARY/16-01-2026/2975.swift
class Solution {
    func maximizeSquareArea(_ m: Int, _ n: Int, _ hFences: [Int], _ vFences: [Int]) -> Int {
        let MOD: Int64 = 1_000_000_007

        var h = hFences
        var v = vFences
        h.append(1)
        h.append(m)
        v.append(1)
        v.append(n)

        h.sort()
        v.sort()

        var hs = Set<Int>()
        for i in 0..<h.count {
            for j in (i + 1)..<h.count {
                hs.insert(h[j] - h[i])
            }
        }

        var best = -1
        for i in 0..<v.count {
            for j in (i + 1)..<v.count {
                let d = v[j] - v[i]
                if hs.contains(d) {
                    best = max(best, d)
                }
            }
        }

        if best == -1 { return -1 }
        return Int((Int64(best) * Int64(best)) % MOD)
    }
}